src/jdk.jpackage/macosx/classes/jdk/jpackage/internal/MacBaseInstallerBundler.java

```diff
@@ -134,6 +135,18 @@ protected void validateAppImageAndBundeler(
                         I18N.getString(
                             "message.app-image-requires-app-name.advice"));
             }
+            if (SIGN_BUNDLE.fetchFrom(params)) {
+                // if signing bundle with app-image, warn user if app-image
+                // is not allready signed.
+                Path launcher = applicationImage.resolve("Contents/MacOS")
+                        .resolve(APP_NAME.fetchFrom(params));
```

Comment:
As far as I remember launcher can be signed, but entire app image might not be signed. So, in this case check will pass, but notarization will fail. I think we should run check on app image itself.

Reply:
jpackage will either copy the launcher from resource unmodified and unsigned, or sign all the sign-able elements in the app-image (including the launcher).